libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -206,12 +212,19 @@ constructor(
             )
         }
 
-        fasterRouteController = FasterRouteController(directionsSession, tripSession, logger)
+        fasterRouteController = FasterRouteController(
+            directionsSession,
+            tripSession,
+            MapboxRouteOptionsProvider(),
```

Comment:
https://github.com/mapbox/mapbox-navigation-android/pull/3184#discussion_r442441483 these topics are kinda related. so moving the question here

right now we're only exposing the ability to override the route refresh route provider. unclear why we're exposing it through MapboxNavigation.
```kotlin
    fun setRouteOptionsProviderOnOffRoute(routeOptionsProvider: RouteOptionsProvider) {
        this.routeOptionsProviderOffRoute = routeOptionsProvider
    }
```

1. why not expose both?
```kotlin
val routeRefreshOptionsProvider = MapboxRouteOptionsProvider()
val fasterRouteOptionsProvider = MapboxRouteOptionsProvider()
```
2. why not expose one? 
``` kotlin
val routeOptionsProvider = MapboxRouteOptionsProvider()
```
3. why not remove setRouteOptionsProviderOnOffRoute?
4. why not keep it as overridable routeRefesh, and private fasterRoute?

all of them can have reasons.. but if there are no reasons to make it overridable, we should go with 3 and share the instances like pablo is saying

Reply:
I wouldn't expose more than require because expose smth easier that hide. I suggest not expose `Faster Route route options provider`, at least for now.